widget/embedded/src/utils/events.ts

```diff
@@ -0,0 +1,34 @@
+import { eventEmitter } from '../services/eventEmitter';
+import { type UiEventData, WidgetEvents } from '../types';
+
+export function emitPreventableEvent(
+  event: {
+    type: UiEventData['type'];
+    payload?: Omit<
+      UiEventData['payload'],
+      'preventDefault' | 'defaultPrevented'
+    >;
+  },
+  action: () => void
+): void {
+  let defaultPrevented = false;
+
+  const extendedPayload = {
+    get defaultPrevented() {
```

Comment:
Not sure why we are exposing this to subscriber. Am I missing something here? If not needed please remove from here and type as well.